internal/kibana/maintenance_window/models.go

```diff
@@ -0,0 +1,388 @@
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/elastic/terraform-provider-elasticstack/generated/kbapi"
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type MaintenanceWindowModel struct {
+	ID             types.String              `tfsdk:"id"`
+	SpaceID        types.String              `tfsdk:"space_id"`
+	Title          types.String              `tfsdk:"title"`
+	Enabled        types.Bool                `tfsdk:"enabled"`
+	CustomSchedule MaintenanceWindowSchedule `tfsdk:"custom_schedule"`
+	Scope          *MaintenanceWindowScope   `tfsdk:"scope"`
+}
+
+type MaintenanceWindowScope struct {
+	Alerting MaintenanceWindowAlertingScope `tfsdk:"alerting"`
+}
+
+type MaintenanceWindowAlertingScope struct {
+	Kql types.String `tfsdk:"kql"`
+}
+
+type MaintenanceWindowSchedule struct {
+	Start     types.String                        `tfsdk:"start"`
+	Duration  types.String                        `tfsdk:"duration"`
+	Timezone  types.String                        `tfsdk:"timezone"`
+	Recurring *MaintenanceWindowScheduleRecurring `tfsdk:"recurring"`
+}
+
+type MaintenanceWindowScheduleRecurring struct {
+	End         types.String `tfsdk:"end"`
+	Every       types.String `tfsdk:"every"`
+	Occurrences types.Int32  `tfsdk:"occurrences"`
+	OnWeekDay   types.List   `tfsdk:"on_week_day"`
+	OnMonthDay  types.List   `tfsdk:"on_month_day"`
+	OnMonth     types.List   `tfsdk:"on_month"`
+}
+
+/* CREATE */
+
+func (model MaintenanceWindowModel) toAPICreateRequest(ctx context.Context) (kbapi.PostMaintenanceWindowJSONRequestBody, diag.Diagnostics) {
+	var diags = diag.Diagnostics{}
+
+	body := kbapi.PostMaintenanceWindowJSONRequestBody{
+		Enabled: model.Enabled.ValueBoolPointer(),
+		Title:   *model.Title.ValueStringPointer(),
+	}
+
+	body.Schedule.Custom.Duration = model.CustomSchedule.Duration.ValueString()
+	body.Schedule.Custom.Start = model.CustomSchedule.Start.ValueString()
+
+	if !model.CustomSchedule.Timezone.IsNull() && !model.CustomSchedule.Timezone.IsUnknown() {
+		body.Schedule.Custom.Timezone = model.CustomSchedule.Timezone.ValueStringPointer()
+	}
+
+	if model.CustomSchedule.Recurring != nil {
+
+		body.Schedule.Custom.Recurring = &struct {
+			End         *string    `json:"end,omitempty"`
+			Every       *string    `json:"every,omitempty"`
+			Occurrences *float32   `json:"occurrences,omitempty"`
+			OnMonth     *[]float32 `json:"onMonth,omitempty"`
+			OnMonthDay  *[]float32 `json:"onMonthDay,omitempty"`
+			OnWeekDay   *[]string  `json:"onWeekDay,omitempty"`
+		}{
+			End:   model.CustomSchedule.Recurring.End.ValueStringPointer(),
+			Every: model.CustomSchedule.Recurring.Every.ValueStringPointer(),
+		}
+
+		if !model.CustomSchedule.Recurring.Occurrences.IsNull() && !model.CustomSchedule.Recurring.Occurrences.IsUnknown() && model.CustomSchedule.Recurring.Occurrences.ValueInt32() > 0 {
```

Comment:
```suggestion
		if utils.IsKnown(model.CustomSchedule.Recurring.Occurrences) {
```

IMO the `ValueInt32() > 0` check shouldn't be here. The attribute is being validated within the schema, ignoring an invalid value here feels like a potential bug (somehow a negative `occurrences` becomes meaningful but we forget to remove this check) rather than a feature.